python_modules/dagster/dagster/components/origin.py

```diff
@@ -0,0 +1,8 @@
+from dagster_shared.record import record
+
+from dagster.components.component.component import Component
+
+
+@record
+class ComponentOrigin:
+    root_component: Component
```

Comment:
what about plain ole components that are not models or dataclasses?

Reply:
arguably this should be `DefsFolderComponent` since the root of the tree will always be one